cmd/oras/internal/option/confirmation.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package option
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/pflag"
+)
+
+// Confirmation option struct.
+type Confirmation struct {
+	Confirmed bool
+}
+
+var scanln func(a ...any) (n int, err error) = fmt.Scanln
+
+// ApplyFlags applies flags to a command flag set.
+func (opts *Confirmation) ApplyFlags(fs *pflag.FlagSet) {
+	fs.BoolVarP(&opts.Confirmed, "yes", "y", false, "do not prompt for confirmation")
+}
+
+func (opts *Confirmation) AskForConfirmation(message string) (bool, error) {
+	if opts.Confirmed {
+		return true, nil
+	}
+
+	for {
+		fmt.Print(message)
+
+		var response string
+		if _, err := scanln(&response); err != nil {
```

Comment:
I wrote in this way because in the unit test, I can easily override the `scanln()` func.
```
var scanln func(a ...any) (n int, err error) = fmt.Scanln
var response string
scanln(&response)
```
Then, in the unit test, I can do things like.
```
reset := func(input string) {
	scanln = func(a ...any) (n int, err error) {
		*a[0].(*string) = input
		return len([]byte(input)), nil
	}
}
reset("yes")
// test for user input "yes"
reset("no")
// test for user input "no"
```
Compare to introducing `w io.Writer` to the func, it is a different way that I found to handle `fmt.Scanln()`. I think it's good, does it make sense to you? @shizhMSFT

Reply:
The way of doing `var scanln` is too hacky. Since it is just for unit test, it does not worth it.

Reply:
> The way of doing `var scanln` is too hacky. Since it is just for unit test, it does not worth it.

No problem, will change to use `w io.Writer` and `fmt.Fscanln`.